### ISSUE TITLE
Version numbers in identifiers

### ISSUE THREAD

**Issue by marcverhagen** [clamsproject/clams-python#82]
It looks like we have two schools of thought: one where the version number is part of the identifier and one where it is not. If we include the version number then having a `version` property is redundant.

**Comment by keighrim** [comment]
Yeah, I was thinking about this too yesterday. At the moment we haven't decided the form of the identifier. I have been just arbitrarily inserting version numbers in the identifier though. I thought we put off this discussion until the further development of the app directory, but it's not happening as quickly as I expected, so I guess now is a good time to formalize the format of the identifier.

**Comment by keighrim** [comment]
tl;dr: Version stays in identifier, the "inconsistency" problem will be fixed by #118, and I don't think the "redundancy" part is a problem. 

---
Based on decision from https://github.com/clamsproject/mmif/issues/153, the identifier string should include the version number so that it can be inserted as a single string into the `app` (*producer*) field when signing a view. And as of #118, the identifier compound is generated programmatically, so that there won't be inconsistency raise by this issue. Lastly, about redundancy, I don't think having version number appended to the app name is redundant, but more of a convenient feature, in that we can enjoy giving a different meaning to the "bare" identifier (without the version suffix), for example; the "latest" version of the app (as in the largest number, not in the "docker" sense).

**Comment by keighrim** [comment]
@marcverhagen also suggested appending `mmif_version` to the identifier string or forcing app developers to use semantic versioning that is loosely coupled with the underlying MMIF version, but I can't remember what was the reason.

**Comment by keighrim** [comment]
Somewhat repeating myself: 
* version stays in identifier
* the "inconsistency" problem is fixed by #118, 
* there is no "redundancy" because `http://apps.clams.ai/some-app` and `http://apps.clams.ai/some-app/v7` may mean different things. 
* having `mmif-version` in the identifier or app_version does seem to be redundant (also haven't heard/seen any follow-up from Marc). 

Closing this issue as resolved.
